src/Mutator/NodeMutationGenerator.php

```diff
@@ -86,27 +85,21 @@ public function __construct(
     /**
      * @return Mutation[]
      */
-    public function generate(Node $node): array
+    public function generate(Node $node): iterable
     {
-        return array_reduce(
-            $this->mutators,
-            function (array $mutations, Mutator $mutator) use ($node): array {
-                return $this->generateForMutator($node, $mutator, $mutations);
-            },
-            []
-        );
+        yield from take($this->mutators)->map(function (Mutator $mutator) use ($node) {
+            yield from $this->generateForMutator($node, $mutator);
+        });
     }
 
     /**
-     * @param Mutation[] $mutations
-     *
-     * @return Mutation[]
+     * @return iterable|Mutation[]
```

Comment:
```suggestion
     * @return iterable<Mutation>
```

Reply:
We discussed in another PR that this isn't an issue, [and PHPStan agrees](https://phpstan.org/r/4d335ef9-f576-430a-82ba-30e10b744fd4). More like `iterable` is redundant here and it should be just `@return Mutation[]`

Reply:
OK, nevemind. I gave it a second though and I think it doesn't matter if PHPStan can dig it all right. It is more important that humans don't have these powers. For human reader `iterable<Mutation>` works best. Definitely quote me on that.

Reply:
I would also argue that when a convention is in place in the codebase, it's better to challenge it in a dedicated PR and change it everywhere :)